apps/frontend/src/contexts/UserProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect, ReactNode } from "react";
+import UserContext from "./userContext";
+import { User } from "../types";
+import axios from "axios";
+import Cookies from "js-cookie";
+
+const UserProvider = ({ children }: { children: ReactNode }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const storedUser = localStorage.getItem("user");
+    if (storedUser) {
+      setUser(JSON.parse(storedUser));
+    }
+    setLoading(false);
+  }, []);
+
+  const setUserAndStopLoading = (user: User) => {
+    localStorage.setItem("user", JSON.stringify(user));
+    setUser(user);
+    setLoading(false);
+  };
+
+  const logout = async () => {
+    setUser(null);
+    localStorage.removeItem("user");
+    Cookies.remove("token");
```

Comment:
Front-end will not have any access to the token. It's not being sent in the http response body. It's in a `Set-Cookie` header that has `httponly = true` meaning that the front-end CANNOT read (not that you shouldn't, but that the browser will not give it to you)